taskcluster/ac_taskgraph/job.py

```diff
@@ -57,13 +56,20 @@ def configure_gradlew(config, job, taskdesc):
     configure_taskdesc_for_run(config, job, taskdesc, job["worker"]["implementation"])
 
 
-def _extract_command(run):
+def _extract_command(run, fetches_dir):
     pre_gradle_commands = run.pop("pre-gradlew", [])
     pre_gradle_commands += [
         _generate_secret_command(secret) for secret in run.get("secrets", [])
     ]
 
-    gradle_command = ["./gradlew"] + run.pop("gradlew")
+    maven_dependencies_dir = path.join(fetches_dir, "android-gradle-dependencies")
+    gradle_repos_args = [
+        "-P{repo_name}Repo=file://{dir}/{repo_name}".format(
+            dir=maven_dependencies_dir, repo_name=repo_name
+        )
+        for repo_name in ("google", "jcenter")
+    ]
+    gradle_command = ["./gradlew"] + gradle_repos_args + ["listRepositories"] + run.pop("gradlew")
```

Comment:
I think there is a limit to the amount of complexity there should be constructing complex commands in taskgraph, vs having that complexity in a script[1]. I think this code is at least nearing that point (if not already past it).

[1] which would incidentally allow having richer error reporting and logging

Reply:
That's a good idea! This script could then be used by both `job.py` and `android-gradle-dependencies.sh`. I'll have a look at it when I convert `android-gradle-dependencies.sh` into a Python script.